x/ratesync/keeper/ibc.go

```diff
@@ -0,0 +1,405 @@
+	}
+
+	var icaMemo types.ICAMemo
+	err := json.Unmarshal([]byte(icaPacket.Memo), &icaMemo)
+	if err != nil {
+		return err
+	}
+
+	if err := k.handleUnsuccessfulAck(ctx, icaPacket, packet, icaMemo); err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeTimeout,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	)
+
+	k.Logger(ctx).Info(
+		"ICA transaction timed out.",
+		"sequence",
+		packet.Sequence,
+		"channel",
+		packet.SourceChannel,
+		"port",
+		packet.SourcePort,
+	)
+
+	return nil
+}
+
+func (k *Keeper) handleUnsuccessfulAck(
+	ctx sdk.Context,
+	icaPacket icatypes.InterchainAccountPacketData,
+	packet channeltypes.Packet, icaMemo types.ICAMemo,
+) error {
+	messages, err := icatypes.DeserializeCosmosTx(k.cdc, icaPacket.GetData())
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrUnknownRequest, "cannot deserialize ica packet data: %v", err)
+	}
+	hc, found := k.GetHostChain(ctx, icaMemo.HostChainId)
+	if !found {
+		return errorsmod.Wrapf(sdkerrors.ErrNotFound, "hostchain not found for id %v", icaMemo.HostChainId)
+	}
+	for _, msg := range messages {
+		switch sdk.MsgTypeURL(msg) {
+		case sdk.MsgTypeURL(&wasmtypes.MsgInstantiateContract{}):
+			// parse the MsgInstantiateContract  to emit the instantiate error event
+			parsedMsg, ok := msg.(*wasmtypes.MsgInstantiateContract)
+			if !ok {
+				k.Logger(ctx).Error(
+					"Could not parse MsgInstantiateContract while handling unsuccessful ack.",
+					"channel", packet.SourceChannel, "sequence", packet.Sequence,
+				)
+				continue
+			}
+			//reset instantiation state so can be retried.
+			switch icaMemo.FeatureType {
+			case types.FeatureType_LIQUID_STAKE_IBC:
+				hc.Features.LiquidStakeIBC.Instantiation = types.InstantiationState_INSTANTIATION_NOT_INITIATED
+			case types.FeatureType_LIQUID_STAKE:
+				hc.Features.LiquidStake.Instantiation = types.InstantiationState_INSTANTIATION_NOT_INITIATED
+			}
+
+			// emit an event for the instantiate confirmation
+			ctx.EventManager().EmitEvent(
+				sdk.NewEvent(
+					types.EventTypeUnsuccessfulInstantiateContract,
+					sdk.NewAttribute(types.AttributeChainID, hc.ChainId),
+					sdk.NewAttribute(types.AttributeSender, parsedMsg.Sender),
+					sdk.NewAttribute(channeltypes.AttributeKeySequence, strconv.FormatUint(packet.Sequence, 10)),
+					sdk.NewAttribute(channeltypes.AttributeKeySrcChannel, packet.SourceChannel),
+					sdk.NewAttribute(channeltypes.AttributeKeyPortID, packet.SourcePort),
+				),
+			)
+		case sdk.MsgTypeURL(&wasmtypes.MsgExecuteContract{}):
+
+			// parse the MsgExecuteContract to emit the execute error event
+			parsedMsg, ok := msg.(*wasmtypes.MsgExecuteContract)
+			if !ok {
+				k.Logger(ctx).Error(
+					"Could not parse MsgExecuteContract while handling unsuccessful ack.",
+					"channel", packet.SourceChannel, "sequence", packet.Sequence,
+				)
+				continue
+			}
+			//Do nothing, relay next epoch
```

Comment:
If it fails, there is no reason for update to not fail on retry.
and the hook will be called every hour, timeout is 1 hour.  ( so it will send the latest value then )